Sources/SwiftParser/TopLevel.swift

```diff
@@ -26,8 +26,13 @@ extension Parser {
   @_spi(RawSyntax)
   public mutating func parseSourceFile() -> RawSourceFileSyntax {
     let items = self.parseTopLevelCodeBlockItems()
+    var extraneousTokens = [RawSyntax]()
+    while currentToken.tokenKind != .eof {
+      extraneousTokens.append(RawSyntax(consumeAnyToken()))
+    }
+    let unexpectedBeforeEof = extraneousTokens.isEmpty ? nil : RawUnexpectedNodesSyntax(elements: extraneousTokens, arena: self.arena)
```

Comment:
What do you mean by an un-parsable decl? If you mean
- A declaration with syntax errors: No, absolutely not. That declaration will get stuck into the the tree with the syntax errors.
- A declaration that’s so broken that the parser doesn’t make any progress trying to parse it: Yes. If we encounter such cases, we should improve recovery inside the parsing logic of that declaration so that the parser makes progress and puts the unexpected tokens into the declarations’s syntax node.

The idea is that `unexpectedBeforeEof` should contain those tokens that were previously `errorToken` of the last `CodeBlockItem` in the top level code block.